rust/src/delta.rs

```diff
@@ -540,8 +550,35 @@ impl DeltaTable {
             self.version(),
         );
 
-        while let PeekCommit::New(new_version, actions) =
-            self.peek_next_commit(self.version()).await?
+        // construct stream yielding (version, bytes)
+        let store = self.storage.clone();
+        let log_stream = futures::stream::iter(self.version() + 1..).map(|i| {
+            let store2 = store.clone();
+            let loc = commit_uri_from_version(i).clone();
+            async move {
+                let out = store2.get(&loc);
+                (i, out.await)
+            }
+        });
+
+        let buf_size = 50; // TODO
+
+        // why is mut needed here and is it OK?
```

Comment:
To answer the question in the comment here, the `mut` is needed because of the `log_buffer.next()` below which will consume from the buffer